File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 7/28/2025, 12:31:47 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = prices[0] , profit = INT_MIN ;
        for(int i=0 ; i<prices.size(); i++){
            buy = min(prices[i] , buy);
            profit = max(profit ,prices[i]-buy);
        }
        return profit ;
    }
};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 7/28/2025, 12:31:52 PM
class Solution {
    public int maxProfit(int[] prices) {
        
        int profit =0;
        int min = prices[0];
        for(int i=1;i<prices.length;i++)
        {
            if(prices[i]<min) min=prices[i];
            profit = Math.max(prices[i]-min,profit);
        }
        return profit;
        
    }
}